subsys/mpsl/clock_ctrl/mpsl_clock_ctrl.c

```diff
@@ -0,0 +1,317 @@
+		return;
+	}
+
+	z_nrf_clock_bt_ctlr_hf_request();
+}
+
+static void m_hfclk_release(void)
+{
+	/* The z_nrf_clock_bt_ctlr_hf_request doesn't count references to HFCLK,
+	 * it is caller responsibility to do not release the clock if there is
+	 * other request pending.
+	 */
+	if (m_hfclk_refcnt < 1) {
+		return;
+	}
+
+	if (atomic_dec(&m_hfclk_refcnt) > 1) {
+		return;
+	}
+
+	z_nrf_clock_bt_ctlr_hf_release();
+}
+
+static bool m_hfclk_is_running(void)
+{
+	/* As of now assume the HFCLK is runnig after the request was put */
+	return ((m_hfclk_refcnt > 0) ? true : false);
+}
+
+static void m_lfclk_calibration_start(void)
+{
+	if (IS_ENABLED(CONFIG_CLOCK_CONTROL_NRF_DRIVER_CALIBRATION)) {
+		z_nrf_clock_calibration_force_start();
+	}
+}
+
+static bool m_lfclk_calibration_is_enabled(void)
+{
+	if (IS_ENABLED(CONFIG_CLOCK_CONTROL_NRF_DRIVER_CALIBRATION)) {
+		return z_nrf_clock_calibration_is_in_progress();
+	} else {
+		return false;
+	}
+}
+
+static int32_t m_lfclk_request(void)
+{
+	struct onoff_manager *mgr = z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_LF);
+	int32_t err;
+
+	/* Workaround for NRFX-6865. The nrf clock control as well as nrfx_clock doesn't enable
+	 * HFXO when LFSYNTH is selected as LFCLK source. Remove the code when nrfx is fixed.
+	 */
+	if (IS_ENABLED(CONFIG_CLOCK_CONTROL_NRF_K32SRC_SYNTH)) {
+		m_hfclk_request();
+	}
+
+	sys_notify_init_spinwait(&m_lfclk_state.cli.notify);
+	err = onoff_request(mgr, &m_lfclk_state.cli);
+	if (err < 0) {
+		return err;
+	}
+
+	atomic_inc(&m_lfclk_state.m_clk_refcnt);
+
+	return 0;
+}
+
+static int32_t m_lfclk_release(void)
+{
+	struct onoff_manager *mgr = z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_LF);
+	int32_t err;
+
+	/* In case there is other ongoing request, cancel it. */
+	err = onoff_cancel_or_release(mgr, &m_lfclk_state.cli);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Workaround for NRFX-6865. The nrf clock control as well as nrfx_clock doesn't enable
+	 * HFXO when LFSYNTH is selected as LFCLK source. Remove the code when nrfx is fixed.
+	 */
+	if (IS_ENABLED(CONFIG_CLOCK_CONTROL_NRF_K32SRC_SYNTH)) {
+		m_hfclk_release();
+	}
+
+	atomic_dec(&m_lfclk_state.m_clk_refcnt);
+
+	return 0;
+}
+
+#elif defined(CONFIG_CLOCK_CONTROL_NRF2)
+
+/* Temporary macro because there is no system level configuration of LFCLK source and its accuracy
+ * for nRF54H SoC series. What more, there is no API to retrieve the information about accuracy of
+ * available LFCLK.
+ */
+#define MPSL_LFCLK_ACCURACY_PPM 500
+
+static const struct nrf_clock_spec m_lfclk_specs = {
+	.frequency = 32768,
+	.accuracy = MPSL_LFCLK_ACCURACY_PPM,
+	/* This affects selected LFCLK source. It doesn't switch to higher accuracy but selects more
+	 * precise but current hungry lfclk source.
+	 */
+	.precision = NRF_CLOCK_CONTROL_PRECISION_DEFAULT,
+};
+
+static void m_lfclk_calibration_start(void)
+{
+	/* This function should not be called from MPSL if CONFIG_CLOCK_CONTROL_NRF2 is set */
```

Comment:
Maybe we can clarify a bit here and for `m_lfclk_calibration_is_enabled()`
```suggestion
	/* This function is not supported when CONFIG_CLOCK_CONTROL_NRF2 is set.
	 * Currently MPSL will never use this API for this configuration.
	 */
```

Reply:
Changed to
```
	/* This function is not supported when CONFIG_CLOCK_CONTROL_NRF2 is set.
	 * As of now MPSL never uses this API in this configuration.
	 */
```